ruby/test/support/shared_queue_assertions.rb

```diff
@@ -1,12 +1,36 @@
 module SharedQueueAssertions
+  class TestCase
```

Comment:
Does it make sense to have a test case class in the actual library, rather than just for test purposes?

Reply:
It's a fake class. Since we're about to support multiple test frameworks, the queue implementations don't know what kind of objects they contain.

However we could expect an interface, and the different implementations would have to wrap whatever test case objects they have in some kind of delegate object to translate to and from string.